### PR TITLE
[Logs] Improve information displayed by the logs agent

### PR DESCRIPTION
**Pull request by mnshdw**
### What does this PR do?

This PR reworks how sources are managed so that the agent can display more relevant information when asking for its `status`. It will now track and report files and docker images being tailed.

### Motivation

Logs agent status was not very useful.

### CODE REVIEW

File: pkg/logs/config/integration_config.go
Comment (tmichelet) [inline pkg/logs/config/integration_config.go]: what exactly should be moved out and why?

File: pkg/logs/config/integration_config.go
Comment (tmichelet) [inline pkg/logs/config/integration_config.go]: same here?

File: pkg/logs/input/container/scanner.go
Comment (tmichelet) [inline pkg/logs/input/container/scanner.go]: ?

**Review by tmichelet — COMMENTED** [review]
Nice cleanup, code looks great, a couple comments to detail or drop.
Could you paste examples of how the status looks like now with different configurations, and could you see why some CI checks failed?

File: pkg/logs/status/status.go
Comment (ajacquemot) [inline pkg/logs/status/status.go]: This `struct` only wraps `sources []*config.LogSource`, is it worth keeping it ?

File: pkg/status/dist/templates/logsagent.tmpl
Comment (ajacquemot) [inline pkg/status/dist/templates/logsagent.tmpl]: I think you could move this logic in the function `Get` from the status package.

File: pkg/logs/config/sources.go
Comment (ajacquemot) [inline pkg/logs/config/sources.go]: I think this definition won't work with auto-discovery, we might need to differentiate config and runtime errors.

File: pkg/logs/config/integration_config.go
Comment (mnshdw) [reply pkg/logs/config/integration_config.go]:
```
	Reg                     *regexp.Regexp
	ReplacePlaceholderBytes []byte
```
I'd consider moving both to avoid mixing concerns, as these are compiled and not part of the actual configuration format. `LogsSource` is a good candidate to hold them.

File: pkg/logs/config/integration_config.go
Comment (mnshdw) [reply pkg/logs/config/integration_config.go]:
```
	TagsPayload     []byte
	ProcessingRules []LogsProcessingRule `mapstructure:"log_processing_rules"`
```
Same reasons: not part of the configuration, and built at run-time.

File: pkg/logs/input/container/scanner.go
Comment (mnshdw) [reply pkg/logs/input/container/scanner.go]: Thanks for spotting this.

File: pkg/logs/config/source.go
Comment (ajacquemot) [inline pkg/logs/config/source.go]: I think there is a lake of information here, what's an input ? I'd rather use `Instance`

**Comment by mnshdw** [comment]
@tmichelet  Here's a sample:
```
==========
Logs-agent
==========

  integration1
  ------------
    Type: file
    Path: /var/log/access.log
    Status: OK

  integration2
  ------------
    Type: tcp
    Port: 10514

  integration3
  ------------
    Type: docker
    Image: test
    Status: Pending

  log
  ---
    Type: file
    Path: /home/vagrant/log/*.log
    Inputs: /home/vagrant/log/hello-world.log /home/vagrant/log/qwe.log /home/vagrant/log/test.log

    Type: file
    Path: /home/vagrant/abc/*.log
    Status: OK

  docker
  ------
    Type: docker
    Image: nginx
    Inputs: a716a8d046e2dd2640ecbafa6fe88c6a61a2c4ff08e45cb793c7f4f0eb5a73a5

    Type: docker
    Label: postgresql
    Status: Pending

    Type: docker
    Status: Pending
```

File: pkg/logs/config/integration_config.go
Comment (tmichelet) [reply pkg/logs/config/integration_config.go]: ok makes sense, would you like to do it in this PR?

File: pkg/logs/config/integration_config.go
Comment (mnshdw) [reply pkg/logs/config/integration_config.go]: 👍

File: pkg/status/dist/templates/logsagent.tmpl
Comment (mnshdw) [reply pkg/status/dist/templates/logsagent.tmpl]: Actually, I'd like to do more in the template as I feel like these concerns don't belong in the code.

File: pkg/logs/status/status.go
Comment (mnshdw) [reply pkg/logs/status/status.go]: It's useful (to me) to pass around sources and abstract away how they are provided / filtered. Also I feel like manipulating a struct makes for easier refactoring than plain arrays. If this PR wasn't already too long I would probably have refactored all the places where we pass arrays of sources. I don't think it's worth worrying to much about it at the moment as this is likely going to change soon with auto config.

File: pkg/logs/config/status.go
Comment (mnshdw) [reply pkg/logs/config/status.go]: I've dropped the array in favor of a single error, but thanks for the tip :)

File: pkg/logs/config/source.go
Comment (mnshdw) [reply pkg/logs/config/source.go]: `Input` felt rather generic to me as a meta-name to describe things we tail (files, containers). If you are strongly against it I'll consider changing it, but I'm not that convinced by `Instance`.

File: pkg/logs/config/integration_config.go
Comment (mnshdw) [reply pkg/logs/config/integration_config.go]: No, we can do that later, this PR is already too big.

File: pkg/logs/config/source.go
Comment (ajacquemot) [reply pkg/logs/config/source.go]: I think it's fine

**Comment by olivielpeau** [comment]
the `reno_linting` CI job is failing but the `releasenote` file is present, it's an issue in the logic of our linter, working on a fix.

I can "force-merge" this if needed.

**Review by tmichelet — DISMISSED** [review]

File: pkg/status/dist/templates/logsagent.tmpl
Comment (ajacquemot) [reply pkg/status/dist/templates/logsagent.tmpl]: 👍

File: pkg/logs/status/status.go
Comment (ajacquemot) [reply pkg/logs/status/status.go]: 👍

**Review by ajacquemot — DISMISSED** [review]

**Review by olivielpeau — APPROVED** [review]
All this code is in Logs' scope, the sample you posted looks nice, approving without thorough review 👍

**Comment by tmichelet** [comment]
@olivielpeau mind merging too? We don't have write access

**Comment by masci** [comment]
@tmichelet you *do* have write access, it's the review from ramen-intake missing :)

**Review by tmichelet — APPROVED** [review]

**Comment by tmichelet** [comment]
🙇

**Review by tmichelet — APPROVED** [review]

**Review by ajacquemot — APPROVED** [review]

**Pull request merged**
